### PR TITLE
feat(community): Added Google Trends Integration 

### PR DESCRIPTION
**Pull request by A-Jatana**
Description:
This PR introduces the Google Trends Tool from SerpAPI, allowing users to use the integrate a tool for Google Trends API with Langchain to retrieve and analyze search interest data. The tool supports viewing trends for specific search queries and exploring related queries to gain deeper insights.

New Files
- `libs/langchain-community/src/tools/google_trends.ts`  
   _Implements the core functionality for the Google Trends Tool, including methods to handle normal queries and related queries._

- `examples/src/tools/google_trends.ts`  
   _Provides an example of how to use the Google Trends Tool in a LangChain integration._

- `libs/langchain-community/src/tools/tests/google_trends.int.test.ts`  
   _Includes integration tests to validate the functionality of the Google Trends Tool for normal and related queries._

- `docs/core_docs/docs/integrations/tools/google_trends.mdx`  
   _Adds documentation with setup instructions and usage examples for the Google Trends Tool._

Updated Files
- `docs/api_refs/blacklisted-entrypoints.json`  
   _Ensures the Google Trends Tool is correctly integrated and avoids exposure of unintended entry points in the API documentation._

- `libs/langchain-community/.gitignore`  
   _Updates to ignore any temporary or generated files related to the Google Trends integration._

- `libs/langchain-community/langchain.config.js`  
   _Configures the LangChain community package to include the Google Trends Tool for proper build and test pipeline integration._

- `libs/langchain-community/package.json`  
   _Adds `google-trends-api` as a dependency and updates metadata to reflect the new tool._

Fixes # [7070](https://github.com/langchain-ai/langchainjs/issues/7070)

Sample Output
```
Query: Coffee

Date From: Nov 19 – 25,, 2023

Date To: Nov 17 – 23, 2024

Min Value: 84

Max Value: 100

Average Value: 91.75471698113208

Percent Change: 3.23%

Trend values: 93, 91, 98, 89, 92, 100, 93, 93, 89, 86, 86, 90, 90, 93, 92, 91, 87, 90, 91, 93, 84, 88, 90, 94, 91, 86, 89, 90, 88, 90, 94, 91, 85, 92, 95, 92, 90, 91, 100, 96, 96, 100, 93, 96, 91, 99, 93, 93, 89, 91, 89, 94, 96

Rising Related Queries: shop coffee tables on sale, coffee loophole recipe, costa coffee hot chocolate death, the coffee table movie, tomoro coffee, javy coffee, ninja coffee machine, dreams coffee, zus coffee, zus coffee near me, blank street coffee, 7 brew coffee, before the coffee gets cold series, ryze, ryze coffee, black sheep coffee, zus coffee menu, the coffee table, mushroom coffee

Top Related Queries: the coffee, coffee shop, coffee near me, coffee table, best coffee, coffee machine, coffee maker, black coffee, coffee cup, coffee shops, coffee house, coffee bean, starbucks coffee, starbucks, iced coffee, coffee beans, espresso, espresso coffee, cold coffee, coffee mug, white coffee, coffee bar, caffeine in coffee, coffe, coffee cups
```

### CODE REVIEW

**Comment by zhaoroger** [comment]
Hi @jacoblee93, could you take a look at our PR? Thank you for your time!

File: examples/src/tools/google_trends.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { GoogleTrendsAPI } from "@langchain/community/tools/google_trends";
```
Comment (jacoblee93) [inline examples/src/tools/google_trends.ts:1]: Can we call this `SERPGoogleTrendsTool`? Just so it's clear this is not officially Google

File: examples/src/tools/google_trends.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+export async function run() {
+  const tool = new GoogleTrendsAPI();
+
+  const res = await tool._call("Monster");
```
Comment (jacoblee93) [inline examples/src/tools/google_trends.ts:6]: Use `.invoke` instead

**Comment by zhaoroger** [comment]
Hi @jacoblee93 thanks for reviewing our PR. I have renamed `GoogleTrendsAPI` to `SERPGoogleTrendsTool` everywhere and changed to `.invoke` as suggested, as well as fixing some lint warnings I missed before.

**Review by jacoblee93 — APPROVED** [review]
Thank you!

**Pull request merged**